File: March 19, 2026.js
function invertMatrix(matrix) {

  let newMatrix = [];
  const letters = [...new Set(matrix.flat())];

  for(let i = 0; i < matrix.length; i++) {
    let row = [];
    for(let j = 0; j < matrix[0].length; j++) {
      if(matrix[i][j] == letters[0]) {
        row.push(letters[1]);
      } else {
        row.push(letters[0]);
      }
    }
    newMatrix.push(row);
  }

  return newMatrix;
}

File: March 19, 2026.py
def invert_matrix(matrix):

    new_matrix = []
    letters = list({letter for row in matrix for letter in row})

    for row in matrix:
        new_row = []
        for letter in row:
            if letter == letters[0]:
                new_row.append(letters[1])
            else:
                new_row.append(letters[0])
        new_matrix.append(new_row)
    return new_matrix

invert_matrix([["a", "b"], ["a", "a"]])